routes/legacy/api.js

```diff
@@ -325,7 +339,10 @@ router.post("/:match_id/pause/", basicRateLimit, async (req, res, next) => {
     )
       matchFinalized = false;
     // Throw error if wrong key or finished match.
-    await check_api_key(matchValues[0].api_key, req.body.key, matchFinalized);
+    await check_api_key(matchValues[0].api_key, keyCheck(req), matchFinalized);
+
+    // Event Emitter
+    const emitter = app.get("eventEmitter");
```

Comment:
instead of declaring the emitter again, would it be possible to just do it once? it would certainly look better and improves code style

Reply:
I actually ran into a small problem. I was really hoping to keep everything contained to `app` itself. That comes with the caveat of having to declare it in each call, if I try and declare it per file, it runs into the issue of `ReferenceError: Cannot access 'app' before initialization`... I think I may just break the event emitter out into it's own class here, and then maybe just do an import instead.